prog/vm/github_runner.rb

```diff
@@ -165,9 +165,16 @@ def before_run
     hop_setup_environment
   end
 
-  def concurrency_available?
+  def quota_available?
     github_runner.installation.project_dataset.for_update.all
-    github_runner.installation.project.runner_core_limit > github_runner.installation.project.github_installations.sum(&:total_active_runner_cores)
+    # In existing Github quota calculations, we compare total allocated cores
+    # with the core limit and allow passing the limit once. This is because we
+    # check quota and allocate VMs in different labels hence transactions and it
+    # is difficult to enforce quotas ih the environment with lots of concurrent
+    # requests. There are some remedies, but it would require some refactoring
```

Comment:
> it is difficult to enforce quotas ih the environment with lots of concurrent requests.

We have already locked the project using `for_update`. Isn't it enforced?

Reply:
Yes, but after we process the label, we commit the transaction, which releases the lock. We pick the VM in another state. If quota is available, multiple progs can move to `allocate_vm`